src/lib/components/chat/SettingsModal.svelte

```diff
@@ -258,92 +267,89 @@
 				.getReader();
 
 			while (true) {
-				const { value, done } = await reader.read();
-				if (done) break;
-
 				try {
+					const { value, done } = await reader.read();
+					if (done) break;
+
 					let lines = value.split('\n');
 
 					for (const line of lines) {
 						if (line !== '') {
-							console.log(line);
 							let data = JSON.parse(line);
-							console.log(data);
-
 							if (data.error) {
 								throw data.error;
 							}
-
 							if (data.detail) {
 								throw data.detail;
 							}
 							if (data.status) {
-								if (!data.digest) {
-									toast.success(data.status);
-
-									if (data.status === 'success') {
-										const notification = new Notification(`Ollama`, {
-											body: `Model '${modelTag}' has been successfully downloaded.`,
-											icon: '/favicon.png'
-										});
-									}
-								} else {
-									digest = data.digest;
+								if (data.digest) {
+									let downloadProgress = 0;
 									if (data.completed) {
-										pullProgress = Math.round((data.completed / data.total) * 1000) / 10;
+										downloadProgress = Math.round((data.completed / data.total) * 1000) / 10;
 									} else {
-										pullProgress = 100;
+										downloadProgress = 100;
 									}
+									modelDownloadStatus[opts.modelName] = {
+										pullProgress: downloadProgress,
+										digest: data.digest
+									};
+								} else {
+									toast.success(data.status);
```

Comment:
suggestion: the idea is not to report toast notification from the queue processor, instead return the final state and handle them from the job completion callback. So UI logic and business logic don't get mixed together

Reply:
Hmm, I can't really think of good ways to report Ollama responses directly to the user other than including in the business logic. If we decide to remove this toast, the user won't know what's going on (at least for a second or two) as it violates visibility of system status heuristics.

Reply:
Hmm, I didn't see any delay in notifications after my download was successful with existing logic. Any specific steps to reproduce?

Reply:
I meant, when you first click on the download button, for a first few seconds, it seems like the button click hasn't been registered.

Reply:
Ah, I think I could improve that tomorrow. Provide faster feedback so the user doesn't try to add the same model again to the queue. As that would give an error message 😅

Reply:
You already did an amazing work, so if anything we can merge this for now and get back to it later :)

Reply:
Pushed the change. It does exactly what you intended. If it's complicated, then I can revert it.

Reply:
@anuraagdjain Thanks for the hard work! It does seem a bit over-engineered imo, let's just revert to the original solution for now and come back to this later! Everything else LGMT!

Reply:
reverted the change. Should be good to merge.